### PR TITLE
Enable ROCm devices to correctly use keras.layers.CuDNNGRU and keras.layers.CuDNNLSTM

### PR DESCRIPTION
**Pull request by JoeNan1**
Using keras.layers.CuDNNGRU and keras.layers.CuDNNLSTM on ROCm devices may introduce accuracy issues due to incorrect weight layout. This modification ensures compatibility with both CUDA and ROCm weight layouts.

### CODE REVIEW

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
           python -m pip install --upgrade pip setuptools
           echo "::set-output name=dir::$(pip cache dir)"
       - name: pip cache
-        uses: actions/cache@v2
+        uses: actions/cache@v4
```
Comment (fchollet) [inline .github/workflows/format.yml]: Please remove these 2 files from the PR

File: keras/layers/rnn/cudnn_gru.py [hunk 2]
```diff
@@ -152,23 +152,34 @@ def _process_batch(self, inputs, initial_state):
...
+        biases = [
+            self.bias[: self.units],                           
+            self.bias[self.units : self.units * 2],            
+            self.bias[self.units * 2 : self.units * 3],        
+            self.bias[self.units * 3 : self.units * 4],        
+            self.bias[self.units * 4 : self.units * 5],        
+            self.bias[self.units * 5 :],                       
+        ]
+        
+        if tf.sysconfig.get_build_info()["is_cuda_build"]:
```
Comment (fchollet) [inline keras/layers/rnn/cudnn_gru.py:174]: ?

Comment (JoeNan1) [reply keras/layers/rnn/cudnn_gru.py:174]: The default layout is compatible with ROCm; here, the CUDA layout is adapted by checking whether the device is CUDA.

File: .github/workflows/format.yml (hunk 1 above)
Comment (JoeNan1) [reply .github/workflows/format.yml]: Reverted to the original state before the changes.
